src/user_interface/keyboard.rs

```diff
@@ -78,24 +76,26 @@ impl Component for Keyboard {
 
 impl Default for Keyboard {
     fn default() -> Self {
-        let menu = action_menu();
-        Self {
-            current: vec!(menu.clone()),
-            menu,
-        }
+        Self { current: vec![Menu::select(MenuChoice::Root)] }
     }
 }
 
 impl Keyboard {
+    pub fn set_menu(&mut self, menu: Menu) {
+        self.current.clear();
+        self.current.push(menu);
+    }
+
     pub fn current(&self) -> Menu {
         self.current.last().unwrap().clone()
     }
 
     pub fn key_pressed(&self, keycode_pressed: &VirtualKeyCode) -> (Vec<Menu>, Option<Action>) {
         let mut current = self.current.clone();
-        if keycode_pressed == &Escape {
-            current.clear();
-            current.push(self.menu.clone());
+        if keycode_pressed == &Escape || keycode_pressed == &Back {
```

Comment:
This reads nicer if you pass an owned VirtualKeyCode instead and compare without reference.